### DISCUSSION TITLE
AttributeError: 'Model' object has no attribute '__pydantic_extra__' during validate assignment

### DISCUSSION THREAD

**Discussion by Vaserti** [fastapi/sqlmodel#767] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from sqlmodel import SQLModel, Field
from pydantic import ConfigDict


class Example(SQLModel, table=True):
    id: int = Field(primary_key=True)
    model_config = ConfigDict(validate_assignment=True)


test = Example(id=1)
test.id = "1"
```

### Description
 
I have this error below:

<img width="1338" alt="image" src="https://github.com/tiangolo/sqlmodel/assets/81198593/760d3aa9-6435-43cc-b414-7b27baee86fc">

### Operating System

macOS

### SQLModel Version

0.0.14

### Python Version

3.11

**Comment by jplhanna** [comment]
I had this issue recently as well, it stopped happening when I dropped the `validate_assignment=True` from the config. Seems like it doesn't play well with how sqlmodel initializes the object.

**Comment by YuriiMotov** [comment]
Works well in current version (0.0.24):

```py
from sqlmodel import SQLModel, Field
from pydantic import ConfigDict, ValidationError


class Example(SQLModel, table=True):
    id: int = Field(primary_key=True)
    model_config = ConfigDict(validate_assignment=True)


test = Example(id=1)
test.id = "123"

print(f"{test=}")

try:
    test.id = "asd"
except ValidationError as e:
    print(e.errors())
```

Output:
```
test=Example(id=123)
[{'type': 'int_parsing', 'loc': ('id',), 'msg': 'Input should be a valid integer, unable to parse string as an integer', 'input': 'asd', 'url': 'https://errors.pydantic.dev/2.11/v/int_parsing'}]
```
